shared-libs/rules-engine/src/pouchdb-provider.js

```diff
@@ -4,13 +4,14 @@
  * Wireup for accessing rules document data via medic pouch db
  */
 
-// TODO work out how to pass in the logger from node/browser
 /* eslint-disable no-console */
 const moment = require('moment');
 const registrationUtils = require('@medic/registration-utils');
 const uniqBy = require('lodash/uniqBy');
 
 const RULES_STATE_DOCID = '_local/rulesStateStore';
+const MAX_QUERY_KEYS = 500;
```

Comment:
great results from this change! seems Pouch should essentially provide thisyes, we have discussed this - so it is nice to see it happen :D
this number was always such a sticking point for me... the results are good, but how did you arrive at `500`?

Reply:
I explained it in the PR description.

> The limit of 500 can be changed. In my sample test with ~10000 reports, it took roughly the same time to get all rows vs query with 500 keys. 

I think there is no winning number here, depending on doc counts, other numbers might be more efficient for some users (for a user with only 5000 reports, the best number would be 100 for example, but for a user with 20000 reports, the best number is 700). 

Do you have a suggestion of a different way of selecting this number?

Reply:
We could plot a curve, but seems like over optimizing.  I'll start by just testing with another heavy dataset and reporting some numbers.

Reply:
Perfect,  that would be super helpful. 
If this is indeed blocking loading for heavy users, I'm inclined to just go with a sub-optimal number. Even if we plot a number, we would probably choose a round one, rather than 763. I think the number should be somewhere between 100 and 1000 (maybe 1000 is already way too much). 
I'm keen to plot a curve.